AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -147,7 +148,18 @@ - (void)_calculateSize
   if (isinf(_constrainedSize.width) == NO && [_attributes.pointSizeScaleFactors count] > 0) {
     _currentScaleFactor = [[self fontSizeAdjuster] scaleFactor];
   }
-  
+
+  // If we do not scale, do exclusion, or do custom truncation, we should just use TextKit for a fast-path.
+  BOOL doesNotScale = _currentScaleFactor == 1;
+  // NOTE: This code does not correctly handle if they set `…` with different attributes.
+  BOOL defaultTruncation = _attributes.avoidTailTruncationSet == nil && [_attributes.truncationAttributedString.string isEqualToString:@"\u2026"];
+  BOOL doesNoExclusion = _attributes.exclusionPaths.count == 0;
+  if (doesNotScale && defaultTruncation && doesNoExclusion) {
+    CGSize baseSize = [_attributes.attributedString boundingRectWithSize:_constrainedSize options:NSStringDrawingUsesLineFragmentOrigin | NSStringDrawingTruncatesLastVisibleLine context:nil].size;
```

Comment:
@Adlai-Holler could you add a comment about the options chosen here, and maybe break them into a local variable for a shorter line length?  I'm not familiar with LineFragmentOrigin / TruncatesLastVisible, and it would be helpful to describe why these options are chosen.